src/crepe/system/RenderSystem.cpp

```diff
@@ -32,16 +32,17 @@ const Camera & RenderSystem::update_camera() {
 		if (!cam.active) continue;
 		const Transform & transform
 			= mgr.get_components_by_id<Transform>(cam.game_object_id).front().get();
-		this->context.set_camera(cam);
-		this->cam_pos = transform.position + cam.offset;
-		return cam;
+		SDLContext::CameraValues cam_val = this->context.set_camera(cam);
+		cam_val.cam_pos = transform.position + cam.data.postion_offset;
+		return cam_val;
 	}
 	throw std::runtime_error("No active cameras in current scene");
 }
 
 bool sorting_comparison(const Sprite & a, const Sprite & b) {
-	if (a.sorting_in_layer < b.sorting_in_layer) return true;
-	if (a.sorting_in_layer == b.sorting_in_layer) return a.order_in_layer < b.order_in_layer;
+	if (a.data.sorting_in_layer < b.data.sorting_in_layer) return true;
+	if (a.data.sorting_in_layer == b.data.sorting_in_layer)
+		return a.data.order_in_layer < b.data.order_in_layer;
```

Comment:
Moet dit niet eigenlijk
```cpp
if (a.data.sorting_in_layer != b.data.sorting_in_layer)
	return a.data.sorting_in_layer < b.data.sorting_in_layer;
if (a.data.order_in_layer != b.data.order_in_layer)
	return a.data.order_in_layer < b.data.order_in_layer;
return false;
```
zijn?

Reply:
in rendersystemtest heb ik een unittest die laat zien dat het goed gesorteed eruit komt. ik zal nog even kijken met jouw oplossing